### PR TITLE
Rename `object-caching` to `object-cache`

### PR DESCRIPTION
**Pull request by tillkruss**
## Summary

WordPress core calls it "Object Cache" not "Object Caching".

Also replaced @dustinrue with @spacedmonkey as code owner, if that's cool?

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

**Review by mitogh — APPROVED** [review]

File: .github/CODEOWNERS
Comment (felixarntz) [inline .github/CODEOWNERS:21]: Let's wait with this change just a little more until we've confirmed. If/once we merge this, we also need to update e.g. the spreadsheet with our PoCs accordingly.

File: .github/CODEOWNERS
Comment (felixarntz) [inline .github/CODEOWNERS:19]: +1 - let's make sure to also update this elsewhere though, e.g. the GitHub label and our focus area organization spreadsheet.

**Review by felixarntz — COMMENTED** [review]
LGTM, however I'd prefer if we can hold off a few more days with merging this - see my comments below.

File: .github/CODEOWNERS
Comment (tillkruss) [reply .github/CODEOWNERS:19]: I've already updated the labels and project in the repo. @bethanylang Where else would this need updating?

File: .github/CODEOWNERS
Comment (tillkruss) [reply .github/CODEOWNERS:21]: It's confirmed.

Screen Shot 2022-01-25 at 8 49 57 AM

File: .github/CODEOWNERS
Comment (mxbclang) [reply .github/CODEOWNERS:19]: @tillkruss Just the focus spreadsheet; updated here: https://docs.google.com/spreadsheets/d/16N5oZ9wE6AkiqMz7b_707eh24vvpjMwsEG67XFAbxy8/edit#gid=0.

**Review by felixarntz — APPROVED** [review]

**Comment by felixarntz** [comment]
Excellent, updated the focus area name in the spreadsheet now as well.

**Pull request merged**
